certify-service/src/main/java/io/mosip/certify/utils/KeyManagerCacheUtil.java

```diff
@@ -0,0 +1,30 @@
+package io.mosip.certify.utils;
+
+import io.mosip.certify.core.exception.CertifyException;
+import io.mosip.kernel.keymanagerservice.dto.AllCertificatesDataResponseDto;
+import io.mosip.kernel.keymanagerservice.dto.CertificateDataResponseDto;
+import io.mosip.kernel.keymanagerservice.service.KeymanagerService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@Slf4j
+@Component("keyManagerCacheUtil")
+public class KeyManagerCacheUtil {
+    @Autowired
+    KeymanagerService keymanagerService;
+
+    @Cacheable(value = "keyidcache", key = "#appId + '-' + #refId")
+    public CertificateDataResponseDto getKid(String appId, String refId) {
+        AllCertificatesDataResponseDto kidResponse = keymanagerService.getAllCertificates(appId, Optional.of(refId));
+        if (kidResponse == null || kidResponse.getAllCertificates() == null || kidResponse.getAllCertificates().length == 0) {
+            log.error("No certificates found for appId: {} and refId: {}", appId, refId);
+            throw new CertifyException("No certificates found");
+        }
+
+        return kidResponse.getAllCertificates()[0];
```

Comment:
Can you create a JIRA to take care of the other certificates for same appId and refId ? This is required when the keys gets rotated. 
Also I am worried, what if the zero's element is an old key and you don't have the latest key in the did document. Anyway we are taking care of that ?